detox/android/detox/src/full/java/com/wix/detox/espresso/matcher/ViewMatchers.kt

```diff
@@ -54,6 +64,42 @@ fun toHaveSliderPosition(expectedValuePct: Double, tolerance: Double): Matcher<V
         }
     }
 
+fun withRegexText(jsRegex: String): Matcher<View> =
+    object: BoundedMatcher<View, TextView>(TextView::class.java) {
+        override fun describeTo(description: Description) {
+            description.appendText("withRegexText($jsRegex)")
+        }
+
+        override fun matchesSafely(view: TextView): Boolean {
+            val text = view.text?.toString()
+            return text?.isMatch(jsRegex) ?: false
```

Comment:
Have you considered replacing the custom `withText` method with this one? Instead of deciding between different matchers (`withRegexText` and Espresso's `withText`) prior to their invocation, you could simply use an if-else in the return value, like this: return `isRegex ? (value?.isMatchJSRegex(expected) ?: false) : value == expected`. Same Q for the other regex-matchers

Reply:
Honestly I'm not sure if it will be better, WDYT?

Reply:
Yeah I believe it would be nicer code-wise to eliminate use of Espresso's `withText`.

My main concern was changing too much - I only briefly looked at the Espresso's implementation so I decided to simply add on regex matching and use their implementation elsewhere, just in case my own implementation would break existing Detox consumers. 

But I think I can open the source code again and have a deeper look to eliminate use of Espresso here, if you think that's okay 👍

Reply:
Generally, that's a great thinking. It makes sense not to disrupt existing usage. 
A thorough implementation, however, poses no risk if done correctly

Reply:
The simpler their implementation, the safer it is to "copy-paste" it and change it according to our needs..
If it adds too much code to our implementation it's suspicious, because it looks like a simple `value == expected` check, then we might miss something in our regex implementation 😅 

I prefer the simpler solution for readability and cleanliness, as you see fit